game-core/src/main/java/games/strategy/engine/chat/HeadlessChat.java

```diff
@@ -73,11 +73,14 @@ public void addMessageWithSound(final String message, final PlayerName from, fin
         .start();
   }
 
-  private void addChatMessage(final String originalMessage, final String from) {
-    final String currentAllText = allText.toString();
-    if (currentAllText.length() > MAX_LENGTH) {
-      allText = new StringBuilder(currentAllText.substring(MAX_LENGTH / 2));
+  private void trimLengthIfNecessary() {
```

Comment:
nit: method ordering is 'wrong' on this one. It's interesting to see a lot of classes where updates occurred by extracting a method, and instead of placing it below, it was easier to place it above. Reading the code gets kinda interesting as you keep scrolling up to find the next method in the method call.